spec/unit/MatrixAction.spec.js

```diff
@@ -0,0 +1,57 @@
+"use strict";
+const MatrixAction = require("../../lib/models/MatrixAction");
+
+describe("MatrixAction", function() {
+    it("should not highlight mentions to text without mentions", () => {
+        let action = new MatrixAction("message", "Some text", "Some text", null);
+        action.formatMentions({
+            "Some Person": "@foobar:localhost"
+        });
+        expect(action.text).toEqual("Some text");
```

Comment:
it's also expected that the function doesn't needlessly generate an HTML component.

Reply:
Now tested for